scripts/release/asset_publish.py

```diff
@@ -277,33 +286,66 @@ def _str2bool(v: str) -> bool:
             )
 
             # Handle specific asset types
-            if asset.type in [assets.AssetType.COMPONENT, assets.AssetType.ENVIRONMENT]:
-                # Assemble command
-                cmd = assemble_command(
-                    asset.type.value, str(asset.spec_with_path),
-                    registry_name, final_version, resource_group, workspace, debug_mode)
-                # Run command
-                run_command(cmd, failure_list, debug_mode)
+            if asset.type == assets.AssetType.COMPONENT:
+                # load component and check if environment exists
+                logger.print(f"spec's path: {asset.spec_with_path}")
+                component = load_component(asset.spec_with_path)
+                env = component.environment
+                ws_env_pattern = re.compile("^(.+):(.+)$")
+                registry_env_pattern = re.compile("^azureml://registries/.+/environments/(.+)/versions/(.+)")
+                match = registry_env_pattern.search(env)
+                if not match:
+                    match = ws_env_pattern.search(env)
+                    if not match:
+                        logger.print(
+                            "Env ID doesnot match with expected workspace or registry pattern"
+                            + f" in {asset.spec_with_path}"
+                        )
+                        failure_list.append(asset)
+                        continue
+
+                # both ws and registry env follows the same grouping
+                env_name = match.group(1)
+                env_version = match.group(2)
+                logger.print(f"Env name:version => {env_name}:{env_version}")
+                logger.print(f"Using final_version {final_version} to check if env registered")
+                try:
+                    mlclient = MLClient(
+                        DefaultAzureCredential(),
+                        subscription_id=subscription_id,
+                        registry_name=registry_name,
+                    )
+                    env = mlclient.environments.get(name=env_name, version=final_version)
+                except Exception as e:
+                    logger.print(f"Fetching component env from registry failed => str{e}")
```

Comment:
`str` would be a literal in the f-string.
```suggestion
                    logger.print(f"Fetching component env from registry failed: {e}")
```